qdrant-landing/content/articles/solving-contexto.md

```diff
@@ -0,0 +1,177 @@
+- Explain what Word2Vec is.
+
+- We can start with random word and just look into the list of words similar to it. If we see a word that is close to the secret word, we use it as a reference. Then we look into the list of words similar to the reference word and so on. If we see the secret word, we stop.
+- This approach works, but it is very slow. It tends to stuck in clusters of words that are similar to each other, which forces us to retrive a lot of words from the model.
+
+- We also can't use tricks from linear algebra to evaluate exact vector based on distances to given points.
+    - First, because we don't know exact word2vec model which was used to sort words.
+    - Second, because we don't know the exact distance to the secret word.
+    - We can only compare distances between words.
+
+The solution should not only account for the most sililar word we found so far, but also consider the distance to the words it found to be unsimilar.
+-->
+
+It's clear that the game is using some kind of Word2Vec model to sort words by their similarity to the secret word.
+
+Word2vec is a method for representing words in a way that captures their meanings and relationships to other words.
+It uses machine learning algorithms to learn the representation of words in a way that captures the meanings of words based on the context in which they appear.
+This means that words with similar meanings will have similar representations, and words that often appear together will also have similar representations.
+The goal of word2vec is to create a compact and efficient representation of words that can be used in natural language processing tasks, such as determining the similarity between words or predicting the next word in a sentence. 
+
+{{< figure src=/articles_data/solving-contexto/cbow-word2vec.webp caption="Word2Vec training architecture">}}
+
+
+Word2vec is one of the first methods used to represent objects in vector space. 
+Currently there are a lot of more sophisticated methods, that can capture meaning of the whole texts, not just a single word.
+But for our purposes word2vec will work just fine.
+
+So, here's the naive approach you've probably already thought of:
+
+We can start with a random word and look into the list of similar words using some Word2Vec model.
+If we see a word closer to the secret word, we use it as a reference and repeat the process.
+
+Although this approach works if we are initially close enough to the secret word, it is generally quite slow and inefficient.
+It tends to get stuck in clusters of words that are similar to each other, forcing us to retrieve many words from the model.
+
+Additionally, using linear algebra techniques to evaluate the exact vector based on distances to given points does not look feasible in this scenario.
+This is because the exact word2vec model used to sort the words is unknown, as is the exact distance to the secret word.
```

Comment:
It was unclear to me what "exact word2vec model" means until I read the prompt put into ChatGPT. Maybe we could replace it with the "original word2vec model"?